src/deltacode/__init__.py

```diff
@@ -48,14 +50,30 @@ class DeltaCode(object):
     the form of File objects) contained in those scans.
     """
     def __init__(self, new_path, old_path, options):
-        self.new = Scan(new_path)
-        self.old = Scan(old_path)
+        self.codebase1 = None
+        self.codebase2 = None
+        try:
+            self.codebase1 = VirtualCodebase(new_path)
+            self.codebase2 = VirtualCodebase(old_path)
+        except Exception as exception:
+            click.secho(exception.message ,fg = "red")
```

Comment:
What exact exceptions can occur here?

Reply:
@majurg , the error faced here are following two types:
1.When the paths of the json files for the are not valid.
2.When the VirtualCode base gets an unnecessary fields during the scanning .
However the 2nd error could be resolved if we use scancode versionn3 during the scanning of the json files.